pandas/core/case_when.py

```diff
@@ -0,0 +1,178 @@
+from __future__ import annotations
+
+from typing import Any
+import warnings
+
+from pandas.util._exceptions import find_stack_level
+
+import pandas as pd
+import pandas.core.common as com
+
+
+def warn_and_override_index(series, series_type, index):
+    warnings.warn(
+        f"Series {series_type} will be reindexed to match obj index.",
+        UserWarning,
+        stacklevel=find_stack_level(),
+    )
+    return pd.Series(series.values, index=index)
+
+
+def case_when(obj: pd.DataFrame | pd.Series, *args, default: Any) -> pd.Series:
+    """
+    Returns a Series based on multiple conditions assignment.
+
+    This is useful when you want to assign a column based on multiple conditions.
```

Comment:
This can be used independently of assigning a column, I suggest this be removed.